models/UserManager.php

```diff
@@ -0,0 +1,289 @@
+                }
+                if(filter_has_var(INPUT_POST, 'role')){
+                    $role = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'role')));
+                }
+                
+                $token = UserManager::token(30);
+
+                $errors = [];
+
+                if(empty($name) || empty($email) || empty($email_again)){
+                    $errors['empty'] = "Veuillez remplier tous les champs";
+                }
+
+                if($email != $email_again){
+                    $errors['different'] = "Les adresses email ne correspondent pas";
+                }
+
+                if(UserManager::email_taken($email)){
+                    $errors['taken'] = "L'adresse email est déjà assignée à un modérateur";
+                }
+
+                if(empty($errors)){
+                    UserManager::add_modo($name,$email,$role,$token);
+                    header("Location:/gestion-des-admins-et-modos");
+                }
+                return $errors;
+            }
+    }
+    
+    function new_verification()
+    {  
+        if(filter_has_var(INPUT_POST, 'submit')){
+            $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+            $token = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'token')));
+
+            $errors = [];
+
+            if(empty($email) || empty($token)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }else if(UserManager::is_modo($email,$token) == 0){
+                $errors['exist'] = "Ce modérateur n'existe pas";
+            }                
+
+            if(empty($errors)){
+                $_SESSION['admin'] = $email;
+                header("Location:/modification-du-mot-de-passe");
+            }
+                return $errors;
+            }
+    }
+    
+    function password_verification(){
+        if(filter_has_var(INPUT_POST, 'submit')){
+            $password = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password')));
+            $password_again = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password_again')));
+
+            $errors = [];
+            
+            if(empty($password) || empty($password_again)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }
+
+            if($password != $password_again){
+                $errors['different'] = "Les mots de passe sont différents";
+            }
+
+            if (preg_match('#^(?=.*[a-z])(?=.*[A-Z]).{8,}$#', $password)) {
+            }
+            else{
+                $errors['non conforme'] = 'Votre mot de passe doit contenir des minuscules et des majuscules et posséder une longueur de 8 caractères au minimum';
+            }	
+                    
+            if(empty($errors)){
+                Usermanager::update_password($password);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class 'Usermanager' in method 'password_verification'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)